File: src/main/java/io/github/asyncronous/exo/item/ItemCore.java
package io.github.asyncronous.exo.item;

import net.minecraft.item.Item;

import io.github.asyncronous.exo.Exoskeleton;
import io.github.asyncronous.exo.core.ICore;

public final class ItemCore
extends Item{
    private final ICore core;

    public ItemCore(ICore core){
        super();
        this.core = core;
        this.setCreativeTab(Exoskeleton.tab);
        this.setTextureName("exo:core/" + core.getName());
        this.setUnlocalizedName("exo.itemCore." + core.getName());
    }
}

File: src/main/scala/io/github/asyncronous/exoskeleton/item/ItemCore.scala
package io.github.asyncronous.exoskeleton.item

import io.github.asyncronous.exoskeleton.Exoskeleton
import io.github.asyncronous.exoskeleton.api.Core
import net.minecraft.item.Item

class ItemCore(val core: Core)
extends Item{
  this.setCreativeTab(Exoskeleton.tab);
  this.setMaxStackSize(1);
  this.setUnlocalizedName("core." + core.getID());
}